### PR TITLE
[REEF-1095] Pass through O.A.R.Tang to hide unnecessarily public APIs

### PR DESCRIPTION
**Pull request by afchung**
JIRA:
  [REEF-1095](https://issues.apache.org/jira/browse/REEF-1095)

### CODE REVIEW

**Comment by afchung** [comment]
@jwang98052 Please have a look to make sure I'm not accidentally hiding any APIs. Thanks!

File: lang/cs/Org.Apache.REEF.Tang/Annotations/Parameter.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ namespace Org.Apache.REEF.Tang.Annotations
     /// <summary>
     /// ParameterAttribute
     /// </summary>
-    [System.AttributeUsage(System.AttributeTargets.Parameter)]
-    public class ParameterAttribute : System.Attribute
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class ParameterAttribute : Attribute
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang/Annotations/Parameter.cs]: Sealed?

File: lang/cs/Org.Apache.REEF.Tang/Exceptions/BindException.cs [hunk 2]
```diff
@@ -21,13 +21,12 @@ namespace Org.Apache.REEF.Tang.Exceptions
...
     {
-        // private static readonly long serialVersionUID = 1L;
-        public BindException(string message)
+        internal BindException(string message)
             : base(message)
         {           
         }
 
-        public BindException(string message, Exception innerException)
+        internal BindException(string message, Exception innerException)
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang/Exceptions/BindException.cs]: sealed?

**Comment by markusweimer** [comment]
I did a pass. Looks good to me.

**Comment by jwang98052** [comment]
Looks good to me.

**Comment by jwang98052** [comment]
I will test and merge

File: lang/cs/Org.Apache.REEF.Tang/Annotations/Parameter.cs (hunk 1 above)
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Annotations/Parameter.cs]: Good catch!

**Comment by jwang98052** [comment]
I will test and merge

**Pull request closed without merging**